cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2_inbound_to_rococo.rs

```diff
@@ -0,0 +1,594 @@
+		assert_bridge_hub_rococo_message_received, assert_bridge_hub_westend_message_accepted,
+		asset_hub_rococo_location,
+		snowbridge_common::{
+			asset_hub_westend_location, bridged_roc_at_ah_westend, create_foreign_on_ah_westend,
+			erc20_token_location, eth_location, fund_on_bh, register_foreign_asset,
+			register_roc_on_bh, set_up_eth_and_dot_pool,
+			set_up_eth_and_dot_pool_on_rococo, set_up_pool_with_wnd_on_ah_westend,
+			snowbridge_sovereign, TOKEN_AMOUNT,
+		},
+	},
+};
+use asset_hub_westend_runtime::ForeignAssets;
+use bridge_hub_westend_runtime::{
+	bridge_common_config::BridgeReward, bridge_to_ethereum_config::EthereumGatewayAddress,
+	EthereumInboundQueueV2,
+};
+use codec::Encode;
+use hex_literal::hex;
+use snowbridge_core::TokenIdOf;
+use snowbridge_inbound_queue_primitives::v2::{
+	EthereumAsset::{ForeignTokenERC20, NativeTokenERC20},
+	Message, XcmPayload,
+};
+use sp_core::{H160, H256};
+use xcm::opaque::latest::AssetTransferFilter::{ReserveDeposit, ReserveWithdraw};
+use xcm_executor::traits::ConvertLocation;
+
+/// Calculates the XCM prologue fee for sending an XCM to AH.
+const INITIAL_FUND: u128 = 500_000_000_000_000;
+
+/// An ERC-20 token to be registered and sent.
+const TOKEN_ID: [u8; 20] = hex!("c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2");
+
+#[test]
+fn send_token_to_rococo_v2() {
+	let relayer_account = BridgeHubWestendSender::get();
+	let relayer_reward = 1_500_000_000_000u128;
+
+	let token: H160 = TOKEN_ID.into();
+	let token_location = erc20_token_location(token);
+
+	let beneficiary_acc_id: H256 = H256::random();
+	let beneficiary_acc_bytes: [u8; 32] = beneficiary_acc_id.into();
+	let beneficiary =
+		Location::new(0, AccountId32 { network: None, id: beneficiary_acc_id.into() });
+
+	let claimer_acc_id = H256::random();
+	let claimer = AccountId32 { network: None, id: claimer_acc_id.into() };
+	let claimer_bytes = claimer.encode();
+
+	// set XCM versions
+	BridgeHubWestend::force_xcm_version(asset_hub_westend_location(), XCM_VERSION);
+	BridgeHubWestend::force_xcm_version(asset_hub_rococo_location(), XCM_VERSION);
+	AssetHubWestend::force_xcm_version(asset_hub_rococo_location(), XCM_VERSION);
+
+	// To pay fees on Rococo.
+	let eth_fee_rococo_ah: xcm::prelude::Asset = (eth_location(), 3_000_000_000_000u128).into();
+
+	// To satisfy ED
+	AssetHubRococo::fund_accounts(vec![(
+		sp_runtime::AccountId32::from(beneficiary_acc_bytes),
+		3_000_000_000_000,
+	)]);
+	BridgeHubWestend::fund_para_sovereign(AssetHubWestend::para_id(), INITIAL_FUND);
+
+	// Register the token on AH Westend and Rococo
+	let snowbridge_sovereign = snowbridge_sovereign();
+	AssetHubRococo::execute_with(|| {
+		type RuntimeOrigin = <AssetHubRococo as Chain>::RuntimeOrigin;
+
+		// Register token on Penpal
+		assert_ok!(<AssetHubRococo as AssetHubRococoPallet>::ForeignAssets::force_create(
+			RuntimeOrigin::root(),
+			token_location.clone().try_into().unwrap(),
+			snowbridge_sovereign.clone().into(),
+			true,
+			1000,
+		));
```

Comment:
The comment seems not correct.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/58/commits/271e84945a4f89494e5b02f5170e5a35ca39797f